### PR TITLE
roachtest: expose flag to unconditionally collect pprof profiles

### LINKED ISSUE #135972: roachtest: add the option to enable continuous (cpu) profiling
Cluster-wide automatic cpu profiling can be enabled via cluster setting [1]. There are multiple use-cases, including (manual) performance debugging, and profile-guided optimization [2].

Ideally, this option could be enabled _metamorphically_ for any (eligible) roachtest. However, at this time, we don't have a general mechanism for _metamorphic_ cluster settings [3]. As a temporary workaround, we could (randomly) enable cpu profiling before a given test is executed (i.e., `TestSpec.Run`). Additionally, we could consider adding a CLI option, e.g., `roachtest run --enable-cpu-profling ...` to force the corresponding cluster settings.

[1] https://www.cockroachlabs.com/docs/stable/automatic-cpu-profiler
[2] https://go.dev/doc/pgo
[3] https://github.com/cockroachdb/cockroach/issues/105807

Jira issue: CRDB-44802

### PR DESCRIPTION
**Pull request by rickystewart**
Epic: CRDB-41952
Closes: #135972
Part of: CRDB-44692
Release note: None

### CODE REVIEW

**Comment by rickystewart** [comment]
Sample job with artifacts: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/18020145?buildTab=artifacts#%2Fkv95%2Fenc%3Dfalse%2Fnodes%3D3%2Frun_1%2Fartifacts.zip!%2F3.pprof_dump;%2Fkv95%2Fenc%3Dfalse%2Fnodes%3D3%2Frun_1%2Fartifacts.zip!%2F2.pprof_dump;%2Fkv0%2Fenc%3Dfalse%2Fnodes%3D3%2Frun_1%2Fartifacts.zip!%2F2.pprof_dump;%2Fkv0%2Fenc%3Dfalse%2Fnodes%3D3%2Frun_1%2Fartifacts.zip!%2F3.pprof_dump;%2Ftpccbench%2Fnodes%3D3%2Fcpu%3D16%2Frun_1%2Fartifacts.zip;%2Ftpccbench%2Fnodes%3D3%2Fcpu%3D16%2Frun_1%2F4.perf;%2Ftpce%2Fc%3D5000%2Fnodes%3D3%2Frun_1%2Fartifacts.zip

File: pkg/cmd/roachtest/test_runner.go [hunk 1]
```diff
@@ -1111,6 +1119,10 @@ func (r *testRunner) runTest(
 		c.grafanaTags = []string{vm.SanitizeLabel(runID), vm.SanitizeLabel(testRunID), vm.SanitizeLabel(c.Name())}
 	}
 
+	if err := maybeSetupCpuProfiling(ctx, t, c); err != nil {
```
Comment (DarrylWong) [inline pkg/cmd/roachtest/test_runner.go:1122]: It seems to me that this is trying to run SQL on the cluster before the test has started? I don't think that would work but lmk if I'm missing something and it works for you.

Instead I think it makes more sense to pass them as cluster settings when `c.Start` is called. If you look at `c.StartE`, it should be straightforward to add something like:

```
	if roachtestflags.ForceCpuProfile {
		settings.ClusterSettings["server.cpu_profile.duration"] = "20s"
		// and so on...
	}
```

This should also do what you want wrt only applying the settings to the system tenant.

Comment (DarrylWong) [reply pkg/cmd/roachtest/test_runner.go:1122]: In the TC run I'm seeing: 

```
failed to set up cpu profiling for cluster [teamcity-18020145-1733261817-02-n4cpu16] - dial tcp 34.138.117.140:26257: connect: connection refused
=== RUN   tpccbench/nodes=3/cpu=16  [metrics: https://go.crdb.dev/roachtest-grafana/teamcity-18020145/tpccbench-nodes-3-cpu-16/1733261946957/1733263746957
failed to set up cpu profiling for cluster [teamcity-18020145-1733261817-04-n4cpu4] - dial tcp 34.73.183.24:26257: connect: connection refused
```

Comment (rickystewart) [reply pkg/cmd/roachtest/test_runner.go:1122]: It looks like it failed to run in one of the tests that I executed in that build, but not for the others.

**Review by srosenberg — APPROVED** [review]
LGTM! 

Ran `roachtest run --force-cpu-profile --cloud gce "^ycsb/A/nodes=3/cpu=32$"`, and confirmed that the cpu profiles for all 3 nodes were successfully fetched.

**Review by DarrylWong — APPROVED** [review]
LGTM

**Comment by rickystewart** [comment]
TFTR's everyone!

bors r=DarrylWong,srosenberg

**Pull request merged**